src/Input/Cabal.hs

```diff
@@ -79,19 +79,38 @@ packagePopularity cbl = mp `seq` (errs, mp)
 -- | Run 'ghc-pkg' and get a list of packages which are installed.
 readGhcPkg :: Settings -> IO (Map.Map PkgName Package)
 readGhcPkg settings = do
+-- On Windows, the `haddock-html` field in `*.conf` files for GHC boot
+-- libraries for GHC >= 9.0 (up to at least GHC 9.6.2) contain errors. For
+-- example, this may be specified:
+--     haddock-html: ${pkgroot}/../../doc/html/libraries/base-4.18.0.0
+-- when the correct specification would be:
+--     haddock-html: ${pkgroot}/../doc/html/libraries/base-4.18.0.0
+-- So, a unique approach is taken to `readGhcPkg` in that case. It assumes that
+-- the `*.conf` files have not been manually corrected.
+#if defined(mingw32_HOST_OS) && __GLASGOW_HASKELL__>=901
```

Comment:
This CPP should have an upper bound, it is fixed in 9.10.1.

Reply:
I'm not sure it's a great idea to bake this logic about some broken bindists into hoogle, someone could build their own bindist which doesn't have this bug and it would be broken.